google/resource_cloudiot_registry.go

```diff
@@ -0,0 +1,411 @@
+
+	call := config.clientCloudiot.Projects.Locations.Registries.Create(parent, deviceRegistry)
+	res, err := call.Do()
+	if err != nil {
+		return err
+	}
+
+	d.SetId(res.Name)
+
+	return resourceCloudiotRegistryRead(d, meta)
+}
+
+func resourceCloudiotRegistryUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	updateMask := make([]string, 0, 5)
+	hasChanged := false
+	deviceRegistry := &cloudiot.DeviceRegistry{}
+
+	d.Partial(true)
+
+	if d.HasChange("event_notification_configs") {
+		hasChanged = true
+		updateMask = append(updateMask, "event_notification_configs")
+		if v, ok := d.GetOk("event_notification_configs"); ok {
+			deviceRegistry.EventNotificationConfigs = expandEventNotificationConfigs(v.([]interface{}))
+		}
+	}
+	if d.HasChange("state_notification_config") {
+		hasChanged = true
+		updateMask = append(updateMask, "state_notification_config")
+		if v, ok := d.GetOk("state_notification_config"); ok {
+			deviceRegistry.StateNotificationConfig = expandStateNotificationConfig(v.(map[string]interface{}))
+
+		}
+	}
+	if d.HasChange("mqtt_config") {
+		hasChanged = true
+		updateMask = append(updateMask, "mqtt_config")
+		if v, ok := d.GetOk("mqtt_config"); ok {
+			deviceRegistry.MqttConfig = expandMqttConfig(v.(map[string]interface{}))
+
+		}
+	}
+	if d.HasChange("http_config") {
+		hasChanged = true
+		updateMask = append(updateMask, "http_config")
+		if v, ok := d.GetOk("http_config"); ok {
+			deviceRegistry.HttpConfig = expandHttpConfig(v.(map[string]interface{}))
+
+		}
+	}
+	if d.HasChange("credentials") {
+		hasChanged = true
+		updateMask = append(updateMask, "credentials")
+		if v, ok := d.GetOk("credentials"); ok {
+			deviceRegistry.Credentials = expandCredentials(v.([]interface{}))
+
+		}
+	}
+
+	if hasChanged {
+		_, err := config.clientCloudiot.Projects.Locations.Registries.Patch(d.Id(),
+			deviceRegistry).UpdateMask(strings.Join(updateMask, ",")).Do()
+		if err != nil {
+			return fmt.Errorf("Error updating registry %s: %s", d.Get("name").(string), err)
+		}
+		for _, updateMaskItem := range updateMask {
+			d.SetPartial(updateMaskItem)
+		}
+	}
+
+	d.Partial(false)
+
+	return resourceCloudiotRegistryRead(d, meta)
+}
+
+func resourceCloudiotRegistryRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	name := d.Id()
+	res, err := config.clientCloudiot.Projects.Locations.Registries.Get(name).Do()
+
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("Registry %q", name))
+	}
+
+	d.Set("name", res.Id)
+
+	if res.EventNotificationConfigs != nil {
+		eventConfigs := make([]map[string]string, len(res.EventNotificationConfigs))
+		for i, config := range res.EventNotificationConfigs {
+			eventConfigs[i] = map[string]string{"pubsub_topic_name": config.PubsubTopicName}
+		}
+		d.Set("event_notification_configs", eventConfigs)
+	}
+	// to keep the data model lean only changes are pushed if not the default is
```

Comment:
Could you clean up the wording in this comment a bit?

Reply:
Done